### ISSUE TITLE
hypertable_relation_size_pretty differs from compressed_hypertable_stats

### ISSUE THREAD

**Issue by sribe-drw** [timescale/timescaledb#2217]
**Relevant system information:**
 - OS: CentOS 7
 - PostgreSQL version 12.3
 - TimescaleDB version 1.7.2
 - Installation method: source

**Describe the bug**
Numbers reported by hypertable_relation_size_pretty vs timescaledb_information.compressed_hypertable_stats, 9MB vs 832MB:

```
# select * from timescaledb_information.compressed_hypertable_stats;

         hypertable_name         | total_chunks | number_compressed_chunks | uncompressed_heap_bytes | uncompressed_index_bytes | uncompressed_toast_bytes | uncompressed_total_bytes | compressed_heap_bytes | compressed_index_bytes | compressed_toast_bytes | compressed_total_bytes
---------------------------------+--------------+--------------------------+-------------------------+--------------------------+--------------------------+--------------------------+-----------------------+------------------------+------------------------+------------------------
 scn.interday_bond_analysis_tsdb |           30 |                       29 | 98 MB                   | 80 MB                    | 0 bytes                  | 178 MB                   | 763 MB                | 69 MB                  | 232 kB                 | 832 MB
(1 row)

# select total_size from hypertable_relation_size_pretty('scn.interday_bond_analysis_tsdb');
 total_size
------------
 9416 kB
```

The original non-timescale source:
```
# select pg_size_pretty(sum(pg_total_relation_size( tablename))) from pg_tables where  tablename like 'interday_bond_analysis_p%';
 pg_size_pretty
----------------
 181 MB
(1 row)
```

So, is this 20:1 compression or 5:1 blow-up?

Breaking out info of compressed chunks seems to match that of compressed_hypertable_stats, we see an awful log of them with the same reported size:

```
# select * from timescaledb_information.compressed_chunk_stats;
            hypertable_name             |               chunk_name                | compression_status | uncompressed_heap_bytes | uncompressed_index_bytes | uncompressed_toast_bytes | uncompressed_total_bytes | compressed_heap_bytes | compressed_index_bytes | compressed_toast_bytes | compressed_total_bytes
----------------------------------------+-----------------------------------------+--------------------+-------------------------+--------------------------+--------------------------+--------------------------+-----------------------+------------------------+------------------------+------------------------
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_5_chunk  | Compressed         | 2440 kB                 | 2232 kB                  | 0 bytes                  | 4672 kB                  | 19 MB                 | 1728 kB                | 8192 bytes             | 20 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_6_chunk  | Compressed         | 2800 kB                 | 2320 kB                  | 0 bytes                  | 5120 kB                  | 21 MB                 | 1968 kB                | 8192 bytes             | 23 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_7_chunk  | Compressed         | 1808 kB                 | 1520 kB                  | 0 bytes                  | 3328 kB                  | 14 MB                 | 1288 kB                | 8192 bytes             | 15 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_8_chunk  | Compressed         | 3072 kB                 | 2568 kB                  | 0 bytes                  | 5640 kB                  | 23 MB                 | 2176 kB                | 8192 bytes             | 26 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_9_chunk  | Compressed         | 2336 kB                 | 1920 kB                  | 0 bytes                  | 4256 kB                  | 18 MB                 | 1656 kB                | 8192 bytes             | 19 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_10_chunk | Compressed         | 2352 kB                 | 1928 kB                  | 0 bytes                  | 4280 kB                  | 18 MB                 | 1672 kB                | 8192 bytes             | 20 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_11_chunk | Compressed         | 2608 kB                 | 2112 kB                  | 0 bytes                  | 4720 kB                  | 20 MB                 | 1856 kB                | 8192 bytes             | 22 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_12_chunk | Compressed         | 2488 kB                 | 2080 kB                  | 0 bytes                  | 4568 kB                  | 19 MB                 | 1760 kB                | 8192 bytes             | 21 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_13_chunk | Compressed         | 2992 kB                 | 2416 kB                  | 0 bytes                  | 5408 kB                  | 23 MB                 | 2120 kB                | 8192 bytes             | 25 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_14_chunk | Compressed         | 1208 kB                 | 968 kB                   | 0 bytes                  | 2176 kB                  | 9328 kB               | 888 kB                 | 8192 bytes             | 10224 kB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_15_chunk | Compressed         | 1344 kB                 | 1096 kB                  | 0 bytes                  | 2440 kB                  | 10 MB                 | 968 kB                 | 8192 bytes             | 11 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_16_chunk | Compressed         | 2232 kB                 | 1816 kB                  | 0 bytes                  | 4048 kB                  | 17 MB                 | 1592 kB                | 8192 bytes             | 19 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_17_chunk | Compressed         | 2456 kB                 | 2224 kB                  | 0 bytes                  | 4680 kB                  | 19 MB                 | 1736 kB                | 8192 bytes             | 20 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_18_chunk | Compressed         | 3600 kB                 | 3088 kB                  | 0 bytes                  | 6688 kB                  | 27 MB                 | 2536 kB                | 8192 bytes             | 30 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_19_chunk | Compressed         | 2824 kB                 | 2376 kB                  | 0 bytes                  | 5200 kB                  | 21 MB                 | 1992 kB                | 8192 bytes             | 23 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_20_chunk | Compressed         | 3928 kB                 | 3168 kB                  | 0 bytes                  | 7096 kB                  | 30 MB                 | 2760 kB                | 8192 bytes             | 33 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_21_chunk | Compressed         | 4440 kB                 | 3624 kB                  | 0 bytes                  | 8064 kB                  | 34 MB                 | 3120 kB                | 8192 bytes             | 37 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_22_chunk | Compressed         | 4632 kB                 | 3680 kB                  | 0 bytes                  | 8312 kB                  | 35 MB                 | 3248 kB                | 8192 bytes             | 39 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_23_chunk | Compressed         | 3624 kB                 | 2992 kB                  | 0 bytes                  | 6616 kB                  | 28 MB                 | 2544 kB                | 8192 bytes             | 30 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_24_chunk | Compressed         | 4784 kB                 | 3816 kB                  | 0 bytes                  | 8600 kB                  | 37 MB                 | 3376 kB                | 8192 bytes             | 40 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_25_chunk | Compressed         | 4448 kB                 | 3640 kB                  | 0 bytes                  | 8088 kB                  | 34 MB                 | 3120 kB                | 8192 bytes             | 37 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_26_chunk | Compressed         | 4520 kB                 | 3600 kB                  | 0 bytes                  | 8120 kB                  | 35 MB                 | 3176 kB                | 8192 bytes             | 38 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_27_chunk | Compressed         | 4888 kB                 | 4032 kB                  | 0 bytes                  | 8920 kB                  | 37 MB                 | 3432 kB                | 8192 bytes             | 41 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_28_chunk | Compressed         | 4632 kB                 | 3712 kB                  | 0 bytes                  | 8344 kB                  | 35 MB                 | 3248 kB                | 8192 bytes             | 39 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_29_chunk | Compressed         | 4296 kB                 | 3480 kB                  | 0 bytes                  | 7776 kB                  | 33 MB                 | 3016 kB                | 8192 bytes             | 36 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_30_chunk | Compressed         | 4840 kB                 | 3912 kB                  | 0 bytes                  | 8752 kB                  | 37 MB                 | 3408 kB                | 8192 bytes             | 40 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_31_chunk | Compressed         | 4840 kB                 | 3912 kB                  | 0 bytes                  | 8752 kB                  | 37 MB                 | 3400 kB                | 8192 bytes             | 40 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_32_chunk | Compressed         | 4856 kB                 | 3888 kB                  | 0 bytes                  | 8744 kB                  | 37 MB                 | 3424 kB                | 8192 bytes             | 40 MB
 scn.interday_bond_analysis_tsdb        | _timescaledb_internal._hyper_4_33_chunk | Compressed         | 4808 kB                 | 3864 kB                  | 0 bytes                  | 8672 kB                  | 37 MB                 | 3376 kB                | 8192 bytes             | 40 MB
```

Is it possible that when chunking results in few rows per chunk, that chunks share storage in some way and this is not taken into account? For reference, the table contains about 700k rows total, and the structure is (bigint, uuid, date, double, double, double, double, double, double, double, double, double, bigint, timestamptz), with three indexes.

**Comment by tylerfontaine** [comment]
Possible duplicate of https://github.com/timescale/timescaledb/issues/1741

**Comment by sribe-drw** [comment]
> Possible duplicate of #1741

Certainly sounds similar, likely related, but I'd highlight the diferences: 1) that it says hypertable_relation_size always returns 32K, while in this case it returns a value that's believable, and 2), I'm seeing compressed_heap_bytes being reported as almost 10x uncompressed_heap_bytes, which seems unlikely, and 3) it states that hypertable_approximate_row_count always returns 0, while in my case it returns 33k, or about 1/20th.

**Comment by sribe-drw** [comment]
I did another one, many more rows (~2B) so each chunk will have a good number of rows. A pretty good clue is that hypertable_relation_size_pretty is reporting a number which equals 32KB * num chunks--960KB. (Which would be 99.998% compression...) But actual size is a >2X increase--I understand floating point numbers are not very compressible, but something seems wrong here with the magnitude of the increase.

**Comment by gayyappan** [comment]
@sribe-drw hypertable_relation_size does not work correctly with compressed hypertables in 1.7.2 . This has been fixed and will be available in 2.0. The information view compressed_chunk_stats gives the  sizes for compressed chunks on a per chunk basis (size before and after compression)

**Comment by gayyappan** [comment]
dup of #1741
